tests/postgres_tests/test_aggregates.py

```diff
@@ -0,0 +1,266 @@
+        values = GeneralTestModel.objects.filter(
+            integer_field=1).aggregate(bitand=BitAnd('integer_field'))
+        self.assertEqual(values, {'bitand': 1})
+
+    def test_bit_and_on_only_false_values(self):
+        values = GeneralTestModel.objects.filter(
+            integer_field=0).aggregate(bitand=BitAnd('integer_field'))
+        self.assertEqual(values, {'bitand': 0})
+
+    def test_bit_and_empty_result(self):
+        GeneralTestModel.objects.all().delete()
+        values = GeneralTestModel.objects.aggregate(bitand=BitAnd('integer_field'))
+        self.assertEqual(values, {'bitand': None})
+
+    def test_bit_or_general(self):
+        values = GeneralTestModel.objects.filter(
+            integer_field__in=[0, 1]).aggregate(bitor=BitOr('integer_field'))
+        self.assertEqual(values, {'bitor': 1})
+
+    def test_bit_or_on_only_true_values(self):
+        values = GeneralTestModel.objects.filter(
+            integer_field=1).aggregate(bitor=BitOr('integer_field'))
+        self.assertEqual(values, {'bitor': 1})
+
+    def test_bit_or_on_only_false_values(self):
+        values = GeneralTestModel.objects.filter(
+            integer_field=0).aggregate(bitor=BitOr('integer_field'))
+        self.assertEqual(values, {'bitor': 0})
+
+    def test_bit_or_empty_result(self):
+        GeneralTestModel.objects.all().delete()
+        values = GeneralTestModel.objects.aggregate(bitor=BitOr('integer_field'))
+        self.assertEqual(values, {'bitor': None})
+
+    def test_bool_and_general(self):
+        values = GeneralTestModel.objects.aggregate(booland=BoolAnd('boolean_field'))
+        self.assertEqual(values, {'booland': False})
+
+    def test_bool_and_empty_result(self):
+        GeneralTestModel.objects.all().delete()
+        values = GeneralTestModel.objects.aggregate(booland=BoolAnd('boolean_field'))
+        self.assertEqual(values, {'booland': None})
+
+    def test_bool_or_general(self):
+        values = GeneralTestModel.objects.aggregate(boolor=BoolOr('boolean_field'))
+        self.assertEqual(values, {'boolor': True})
+
+    def test_bool_or_empty_result(self):
+        GeneralTestModel.objects.all().delete()
+        values = GeneralTestModel.objects.aggregate(boolor=BoolOr('boolean_field'))
+        self.assertEqual(values, {'boolor': None})
+
+    def test_string_agg_requires_delimiter(self):
+        with self.assertRaises(TypeError):
+            GeneralTestModel.objects.aggregate(stringagg=StringAgg('char_field'))
+
+    def test_string_agg_charfield(self):
+        values = GeneralTestModel.objects.aggregate(stringagg=StringAgg('char_field', delimiter=';'))
+        self.assertEqual(values, {'stringagg': 'Foo1;Foo2;Foo3;Foo4'})
+
+    def test_string_agg_empty_result(self):
+        GeneralTestModel.objects.all().delete()
+        values = GeneralTestModel.objects.aggregate(stringagg=StringAgg('char_field', delimiter=';'))
+        self.assertEqual(values, {'stringagg': None})
```

Comment:
@akaariai, my opinion is that `ArrayAgg` is a special case, where we need to always return an array even if array will be empty.
So, maybe `StringAgg` should always return a string?